### PR TITLE
Add code-saver extension

### PR DESCRIPTION
**Pull request by Gnosnay**
## Description

Store your code snippets and export them freely when you need it.

Pure local managed Snippet store. You should 
1. Pick one local data store first. 
2. Create one Library then
3. Create snippet that you need.
4. Brower the snippts you created.

## Screencast

code-saver-1
code-saver-2
code-saver-3
code-saver-4
code-saver-5
code-saver-6
code-saver-7
code-saver-8
code-saver-9
code-saver-10

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋

I gave it a new spin and couldn't pass `Library` field, no options - I'm not sure what to do here.

code-saver 2023-11-27 at 09 49 58

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by Gnosnay** [comment]
> Hi 👋
> 
> I gave it a new spin and couldn't pass `Library` field, no options - I'm not sure what to do here.
> 
> code-saver 2023-11-27 at 09 49 58
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

@pernielsentikaer hi in this page, can press `cmd + k` to add label and library.
i think i can add one more tooltip here to advise user to help themsleves

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Ah, that makes sense! I guess others might be confused too.

Regarding the labels, I selected #006699 for both of them, but when I wanted to add them to the snippet, they appeared in two random colors. Is that intended?

https://github.com/raycast/extensions/assets/38033107/00c398dd-7fe0-41ac-bed2-1228f1ac46e0

**Comment by Gnosnay** [comment]
> Ah, that makes sense! I guess others might be confused too.
> 
> Regarding the labels, I selected #6699 for both of them, but when I wanted to add them to the snippet, they appeared in two random colors. Is that intended?
> 
>  Raycast-FlC7hFOZ.mp4

no it is one bug. thx for pointing it out.  thank you so much

File: extensions/code-saver/src/components/search-snippet/item-actions.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+              labelsUUID: snippet.labels.map((l) => l.uuid),
+              libraryUUID: snippet.library.uuid,
+              onUpdateSuccess,
+            }}
+          />
+        }
+      />
+      <Action
+        icon={Icon.DeleteDocument}
+        title="Delete This Snippet"
```
Comment (pernielsentikaer) [inline extensions/code-saver/src/components/search-snippet/item-actions.tsx:43]:
```suggestion
        title="Delete This Snippet"
        style={Action.Style.Destructive}
```

File: extensions/code-saver/src/components/creation/label-entry.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      }
+      navigationTitle={uuid ? "Update Label" : "Create New Label"}
+    >
+      <Form.Description
+        text="Give title and color hex for this label.
+Leaving color hex blank will generate random one color."
+      />
+      <Form.TextField
+        id="title"
+        title="Label title"
```
Comment (pernielsentikaer) [inline extensions/code-saver/src/components/creation/label-entry.tsx:102]:
```suggestion
        title="Label Title"
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- There is flicker in both command when you enters them, is there a way not to have this flicker?

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by Gnosnay** [comment]
> Hi 👋
> 
> Thanks for your contribution 💪
> 
> I have now tested your extension, and I have some feedback ready for you:
> 
> * There is flicker in both command when you enters them, is there a way not to have this flicker?
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

hmm actually the flicker is one loading page how db schema upgrading or dependency downloading.

it is too fast so that the loading page is like one flicker.

do you have any recommendation about this?

**Review by pernielsentikaer — APPROVED** [review]
I'd say we go with this and see if we can improve it in v2(or 3) 🙂

**Pull request merged**